src/amo/components/UserCollection/index.js

```diff
@@ -31,7 +31,7 @@ export const UserCollectionBase = (props: InternalProps): React.Node => {
   const linkProps = {};
   let numberText;
 
-  if (loading) {
+  if (loading || numberOfAddons === null) {
```

Comment:
Shouldn't we try to fix the problem here instead? https://github.com/mozilla/addons-frontend/blob/cabfc342cdbd1fed6c9907051cdedceb8998971a/src/amo/pages/CollectionList/index.js#L65-L93

Reply:
I considered doing that, but it strikes me as more appropriate for the `UserCollection` component to be able to handle, and understand, a collection that is loading, than to require the `CollectionList` component to understand when a collection is loading, and then pass that information into `UserCollection`. I think it's a bit of a six-of-one-half-a-dozen-of-another situation, but if a component needs to understand when a collection might be loading, it seems that the `UserCollection` is a more logical place for that logic, as opposed to the `CollectionList` component.

I originally took an entirely different approach, which was to simply set `numberOfAddons` to `0`, instead of `null`, in the reducer, and that fixed the problem, but that seems more like a bandaid than a fix to me.

Reply:
> I originally took an entirely different approach, which was to simply set `numberOfAddons` to `0`, instead of `null`, in the reducer, and that fixed the problem, but that seems more like a bandaid than a fix to me.

Well, that does not sound bad to me and that would match the Flow definition as well.

Reply:
It's true that it matches the Flow definition, but I'm not sure that the Flow definition is correct. The reason I see it as a band-aid, is because it does fix this particular error, which is that we have an `invariant` which is checking `numberOfAddons` for a number, but it doesn't address the bigger issue which is that the code for `UserCollection` does not seem to account for the fact that a collection could be in a loading state. This wouldn't normally happen, but as we've seen with this issue, it's certainly possible.

This is why I am less comfortable just setting `numberOfAddons` to `0` when it is loading, but it's true it would fix the issue.

I can take either route. Do you strongly prefer that I just set `numberOfAddons` to `0` when it is loading?

Reply:
I see what you mean. No need to change `numberOfAddons`.

I think, what bothers me with the current patch (which is r+able, that isn't the problem) is that we have 2 props to define a same behavior. Maybe we should get rid of `loading` and use `numberOfAddons: number|null` only?

Reply:
Yes, I can see how that's annoying. I updated the patch to just use `numberOfAddons`. I'm a bit concerned with the indirection of it, but I added a comment to explain that `numberOfAddons` is `null` when the collection is in a loading state, so I think that should cover it. I tried defining a `loading` const at the top, which was just equal to `numberOfAddons === null`, but Flow didn't quite understand what I was doing at lines 46 and 47, so I stuck with just using `numberOfAddons === null` in a couple of places. 

Take a look at the latest and see what you think.